/*
 *
 * Copyright 2019 Asylo authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */

syntax = "proto2";

package asylo.sgx;

// This file defines the messages required by Intel attestation service client
// to access Intel ECDSA attestation service APIs.
//
// The specification of Intel ECDSA attestation service APIs can be found at:
// https://api.portal.trustedservices.intel.com/documentation

// JSON-encoded TCB Info with its signature.
message SignedTcbInfo {
  // SGX TCB Info encoded as JSON string. Required.
  optional string tcb_info_json = 1;

  // Signature of TcbInfo over the contents of |tcb_info_json|. Required.
  optional bytes signature = 2;
}

// SGX Certificate Authority (CA) Type.
enum SgxCaType {
  SGX_CA_TYPE_UNKNOWN = 0;

  // Intel SGX Processor CA.
  PROCESSOR = 1;

  // Intel SGX Platform CA.
  PLATFORM = 2;
}
